Eatery/Views/Eateries/FilterBar.swift

```diff
@@ -145,10 +145,41 @@ class FilterBar: UIView {
 
     @objc func buttonPressed(sender: UIButton) {
         sender.isSelected.toggle()
-
+        
         if sender.isSelected {
             let filter = displayedFilters[sender.tag]
             selectedFilters.insert(filter)
+
+            // check if filters match that of campus or ctown (taking into account
+            // that nearest first is shared by both)
+            let isCTown = displayedFilters.allSatisfy({ displayedFilter -> Bool in
+                Filter.categoryFilters.contains(displayedFilter) || displayedFilter == Filter.nearest
+            })
+            let payload : Payload!
+            switch filter{
+            case .nearest:
+                payload = isCTown ? CollegetownFilterPressPayload() : NearestFilterPressPayload()
```

Comment:
Is `CollegetownFilterPressPayload` the event when the "nearest" filter is pressed when on the CollegetownEateriesVC? If so this name is misleading.